src/Components/Server/src/ComponentHub.cs

```diff
@@ -69,7 +72,7 @@ public override Task OnDisconnectedAsync(Exception exception)
         {
             // If the CircuitHost is gone now this isn't an error. This could happen if the disconnect
             // if the result of well behaving client hanging up after an unhandled exception.
-            var circuitHost = GetCircuit();
+            var circuitHost = _circuitHandleRegistry.GetCircuit(Context.Items, CircuitKey);
```

Comment:
Couldn't we configure Context.Items instead of introducing the registry?

Reply:
The original intent here was to have an injected type that we could mock in the tests to avoid actually have to created a circuit instance. This originally started out as a singleton service with its own internal dictionary. However, that doesn't work because `Context.Items` is set up to correctly exist at a connection scope so I set up the registry as a wrapper around `Context.Items` with the ability to override in the code.

It's not possible for us to configure `Context.Items` directly in the test since the setting/getting happens entirely within the `StartCircuit` invocation so it's not possible to set up/tear down ahead of time.